### DISCUSSION TITLE
Invalid Eye Gaze Data

### DISCUSSION THREAD

**Discussion by Alfredelfving** [microsoft/MixedRealityToolkit-Unity#9629] [Q&A]
[I](url) am using HoloLens 2 and I have an app that is collecting eye gaze data through MRTK.
I save the data to a database, but I cannot find any information of if the data is valid or not.
The data seems to be very off for some reason, for example, I tried to sit in my chair and spin around 360 degrees, and so the eye gaze direction data for this should have normals that point out of all directions.
The data I get instead is very small float, this is a smal example of some of the data:

`Id          CompletionId GazeOriginX   GazeOriginY   GazeOriginZ   GazeDirectionsX GazeDirectionsY GazeDirectionsZ GazeTimeStamp
----------- ------------ ------------- ------------- ------------- --------------- --------------- --------------- ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
239711      34           -0,007848     0,058328      0,097419      0,067373        -0,178099       0,981704        08:24:59.9105983
239712      34           -0,007848     0,058328      0,097419      0,067373        -0,178099       0,981704        08:24:59.9105983
239713      34           -0,007848     0,058328      0,097419      0,067373        -0,178099       0,981704        08:24:59.9105983
239714      34           -0,007848     0,058328      0,097419      0,067373        -0,178099       0,981704        08:24:59.9105983
240264      34           -0,007602     0,058315      0,097463      0,065918        -0,178268       0,981772        08:25:00.0106010
240265      34           -0,007359     0,058304      0,097513      0,064537        -0,178426       0,981835        08:25:00.0106011`

see the file this [text paste](https://justpaste.it/6v5sx) for full database table records...

How do I validate this data? The gaze origin seems to be the most off, since the distance if I walk lets say 5 meters does not correspond to euclidean distance of 5 meter with the data at all. All the gaze origin data is < 1, which makes no sense if I would walk several meters...

I have tried to recalibrate with eye calibration, cleaned the hololens fully,

**Comment by keveleigh** [comment]
Which Unity version are you using? And are you using legacy Windows Mixed Reality or Windows XR Plugin or OpenXR?
